test/sqllogictest/string.slt

```diff
@@ -734,6 +734,425 @@ SELECT lpad('str', 5, 'đẹp')
+----
+true
+
+query B
+SELECT '\A' ILIKE '\\A'
+----
+true
+
+query B
+SELECT 'A' ILIKE '\\A'
+----
+false
+
+query B
+SELECT '_' ILIKE '\_'
+----
+true
+
+query B
+SELECT '\' ILIKE '\\'
+----
+true
+
+query B
+SELECT 'A\A' ILIKE '_\\_'
+----
+true
+
+query B
+SELECT '__' ILIKE '_\\_'
+----
+false
+
+query B
+SELECT '\_' ILIKE '\\\_'
+----
+true
+
+query B
+SELECT '\\' ILIKE '\\'
+----
+false
+
+query B
+SELECT '\\' ILIKE '\\_'
+----
+true
+
+query B
+SELECT '\\' ILIKE '_\\'
+----
+true
+
+query B
+SELECT 'A\' ILIKE '_\\'
+----
+true
+
+query B
+SELECT '%' ILIKE '\%'
+----
+true
+
+query B
+SELECT 'ABC' ILIKE '\AB%'
+----
+true
+
+query B
+SELECT 'ABC' ILIKE '\AB_'
+----
+true
+
+query B
+SELECT 'ABC' ILIKE '%B\C'
+----
+true
+
+query B
+SELECT 'ABC' ILIKE '_B\C'
+----
+true
+
+query B
+SELECT 'TEST' ILIKE 'TE\ST'
+----
+true
+
+query B
+SELECT '_漢' ILIKE '\__'
+----
+true
+
+query B
+SELECT '漢漢' ILIKE '漢\漢'
+----
+true
+
+query B
+SELECT '_漢' ILIKE '\_\漢'
+----
+true
+
+query B
+SELECT 'TEST' ILIKE 'TE%'
+----
+true
+
+query B
+SELECT 'TEST' ILIKE '%E%'
+----
+true
+
+query B
+SELECT 'TEST' ILIKE '%e%'
+----
+true
+
+query B
+SELECT 'TEST' ILIKE 'TES_'
+----
+true
+
+query B
+SELECT 'TEST' ILIKE 'TE_%'
+----
+true
+
+query B
+SELECT 'TEST' ILIKE 'TE_'
+----
+false
+
+query B
+SELECT 'TEST' ILIKE '%'
+----
+true
+
+query B
+SELECT 'TEST' ILIKE '%R'
+----
+false
+
+query B
+SELECT 'T' ILIKE '\_'
+----
+false
+
+query B
+SELECT 'T' ILIKE '\%'
+----
+false
+
+query B
+SELECT 'TE_T' ILIKE 'TE\_T'
+----
+true
+
+query B
+SELECT 'TE\AT' ILIKE 'TE\_T'
+----
+false
+
+query B
+SELECT 'TES%T' ILIKE 'TES\%T'
+----
+true
+
+query B
+SELECT 'TES\AT' ILIKE 'TES\%T'
+----
+false
+
+query B
+SELECT 'T' ILIKE '_'
+----
+true
+
+query B
+SELECT 'TE' ILIKE '_'
+----
+false
+
+query B
+SELECT 'TE' ILIKE '_%'
+----
+true
+
+query B
+SELECT 'T' ILIKE '_%'
+----
+true
+
+query B
+SELECT '' ILIKE '_%'
+----
+false
+
+query B
+SELECT 'TE' ILIKE '%_'
+----
+true
+
+query B
+SELECT '' ILIKE '%_'
+----
+false
+
+query B
+SELECT 'T' ILIKE '%_'
+----
+true
+
+query B
+SELECT 'TEST' ILIKE '_ES_'
+----
+true
+
+query B
+SELECT '' ILIKE '__'
+----
+false
+
+query B
+SELECT 'A' ILIKE 'T_'
+----
+false
+
+query B
+SELECT 'A' ILIKE '_T'
+----
+false
+
+query B
+SELECT 'TEST' ILIKE '_E%'
+----
+true
+
+query B
+SELECT 'TEST' ILIKE '_E\%'
+----
+false
+
+query B
+SELECT 'TES_' ILIKE '%S\_'
+----
+true
+
+query B
+SELECT 'TES%' ILIKE '%S\%'
+----
+true
+
+query B
+SELECT 'TES_' ILIKE '_ES\_'
+----
+true
+
+query B
+SELECT 'TES%' ILIKE '_ES\%'
+----
+true
+
+query B
+SELECT 'TEST' ILIKE '%S_'
+----
+true
+
+query B
+SELECT 'TEST' ILIKE '%S\_'
+----
+false
+
+query B
+SELECT 'TEST' NOT ILIKE '%E%'
+----
+false
+
+query B
+SELECT 'TEST' NOT ILIKE 'TES_'
+----
+false
+
+query B
+SELECT 'TEST' NOT ILIKE 'TeS_'
+----
+false
+
+query B
+SELECT 'TEST' NOT ILIKE 'TE_'
+----
+true
+
+# Test bad escaping.
+
+query error unterminated escape sequence in LIKE
+SELECT 'a' ILIKE '\'
+
```

Comment:
I think test coverage will be slightly improved if you lift Cockroach's dedicated `ILIKE` tests (https://github.com/cockroachdb/cockroach/blob/d2f7fbf5dd1fc1a099bbad790a2e1f7c60a66cc3/pkg/sql/sem/tree/testdata/eval/like#L1215-L1313) rather than adapting the existing `LIKE` tests. They focus more on testing the casing rather than the escaping behavior; since the escaping stuff should share all the code with normal `LIKE` probably doesn't need to be tested into too much detail.